src/rust_connection/stream.rs

```diff
@@ -300,80 +447,79 @@ impl ReadFD for Stream {
             use std::io::Read;
             // No FDs are read, so nothing needs to be done with fd_storage
             let _ = fd_storage;
-            match self.inner {
-                StreamInner::TcpStream(ref mut stream) => stream.read(buf),
+            loop {
+                let read_result = match self.inner {
+                    DefaultStreamInner::TcpStream(ref stream) => {
+                        let mut stream = stream;
```

Comment:
`(&mut stream).read(buf)` does not work beacuse `stream: &TcpStream` is not `mut`. `(&mut &*stream).read(buf)?` works because it is creating a temporary `&TcpStream`, which is mutable, so I'll use that with a comment.